pkg/addonmanager/controllers/agentdeploy/default_sync.go

```diff
@@ -35,21 +53,32 @@ func (s *defaultSyncer) sync(ctx context.Context,
 		return addon, nil
 	}
 
-	deployWork, _, err := s.controller.buildManifestWorks(ctx, s.agentAddon, installMode, deployWorkNamespace, cluster, addon)
+	deployWorks, _, err := buildManifestWorks(ctx, s.agentAddon, installMode, deployWorkNamespace, cluster, addon)
 	if err != nil {
 		return addon, err
 	}
 
-	// deployWork is nil only in the case that there is no object from addon manifests. need to delete the deployWork.
-	if deployWork == nil {
-		err = deleteWork(ctx, s.controller.workClient, deployWorkNamespace, deployWorkName)
+	requiredWorkNames := sets.NewString()
```

Comment:
I noticed that the logic of comparing current work and required work and updating is almost the same as hosted_sync, is it possible to extract a common function like updateWorks?